tracer/src/Datadog.Trace/Debugger/DebuggerManager.cs

```diff
@@ -36,19 +36,25 @@ internal class DebuggerManager
                     ExceptionReplaySettings.FromDefaultSource()),
                 LazyThreadSafetyMode.ExecutionAndPublication);
 
-        private static volatile bool _discoveryServiceReady;
-        private readonly SemaphoreSlim _semaphore;
-        private readonly TaskCompletionSource<bool> _processExit = new(TaskCreationOptions.RunContinuationsAsynchronously);
+        private readonly object _syncLock;
+        private readonly TimeSpan _diDebounceDelay;
+        private readonly TaskCompletionSource<bool> _processExit;
+        private volatile bool _isDebuggerEndpointAvailable;
         private int _initialized;
+        private CancellationTokenSource? _diDebounceCts;
```

Comment:
We shouldn't use `CancellationTokenSource` in paths that can be hit during shutdown, because it can cause crashes. We generally have to use `TaskCompletionSource` instead (even though it's a bit of a pain)

Reply:
Thanks for pointing that. Fixed in f3d6f4c949964231419ab232c0b5939abb08ac8e